packages/wallet-ui/src/pages/account-deposit/AccountDeposit.tsx

```diff
@@ -118,26 +119,37 @@ export class AccountDeposit extends React.Component<
   render() {
     const { walletState, deposit, history, user } = this.props;
     const { provider } = this.context;
-    const { ethereumBalance, error, status, tokenAddresses } = walletState;
+    const { error, status, tokenAddresses } = walletState;
     const { amount, loading, depositCaseVariables, tokenAddress } = this.state;
     const {
       halfWidget,
       header,
       headerDetails,
       ctaButtonText
     } = depositCaseVariables;
+
     return (
       <WidgetScreen header={header} half={halfWidget} exitable={false}>
         <form>
           <div className="details">{headerDetails}</div>
           <FormInput
-            label={<BalanceLabel available={formatEther(ethereumBalance)} />}
+            label={
+              <BalanceLabel
+                available={formatEther(
+                  (tokenAddresses[0] && tokenAddresses[0].walletBalance) || Zero
```

Comment:
Sounds like we'd want a `getFormattedBalanceFrom(tokenAddresses, index)` utility function to favor DRY, so we don't have this `|| Zero` fallback everywhere.

Reply:
done!